distributedlog-core/src/main/java/com/twitter/distributedlog/EnvelopedEntry.java

```diff
@@ -254,21 +253,22 @@ private void read(DataInputStream in) throws IOException {
         Payload() {
         }
 
-        Payload(int length, byte[] payload) {
+        Payload(int length, ByteBuffer payload) {
             this.length = length;
             this.payload = payload;
             this.ready = true;
         }
 
         private void write(DataOutputStream out) throws IOException {
             out.writeInt(length);
-            out.write(payload, 0, length);
+            out.write(payload.array(), 0, length);
         }
 
         private void read(DataInputStream in) throws IOException {
             this.length = in.readInt();
-            this.payload = new byte[length];
-            in.readFully(payload);
+            this.payload = ByteBuffer.wrap(new byte[length]);
+            //TODO: Fix this
```

Comment:
@leighst The existing Stream based implementation in buffer.java involved an avoidable byte array copy. This change request is to fix this. We realized that this can be done in a better way. I/@jiazhai will update the pull request with the latest changes.